akka-docs-new/src/test/java/jdocs/tutorial_2/DeviceInProgress.java

```diff
@@ -0,0 +1,26 @@
+package jdocs.tutorial_2;
+
+import java.util.Optional;
+
+import jdocs.tutorial_2.Device.ReadTemperature;
+import jdocs.tutorial_2.Device.RecordTemperature;
+import jdocs.tutorial_2.Device.RespondTemperature;
+import jdocs.tutorial_2.Device.TemperatureRecorded;
+
+class DeviceInProgress1 {
+
+    //#read-protocol-1
+    final static class ReadTemperature {
```

Comment:
The user is expected to place these classes into their `Device.java`. We just need the namespacing to be able to show different in-progress variations of the same class, the reader won't have that problem (though I agree it can get somewhat confusing)